addons/test_mail_full/tests/test_portal.py

```diff
@@ -240,18 +238,16 @@ def setUpClass(cls):
                         sort=True),
              parsed_url[4]))
 
-    def assert_URL(self, url, expected_path, expected_fragment_params=None, expected_query=None):
-        """Asserts that the URL has the expected path and if set, the expected fragment parameters and query."""
+    def assert_URL(self, url, expected_path, expected_query=None, expected_fragment=None):
+        """Asserts that the URL has the expected path and if set, the expected query and fragment."""
         parsed_url = url_parse(url)
-        fragment_params = url_decode(parsed_url.fragment)
         self.assertEqual(parsed_url.path, expected_path)
-        if expected_fragment_params:
-            for key, expected_value in expected_fragment_params.items():
-                self.assertEqual(fragment_params.get(key), expected_value,
-                                 f'Expected: "{key}={expected_value}" (for path: {expected_path})')
         if expected_query:
             self.assertEqual(expected_query, parsed_url.query,
                              f'Expected: query="{expected_query}" (for path: {expected_path})')
+        if expected_fragment:
+            self.assertIn(expected_fragment, parsed_url.fragment,
+                             f'Expected: fragment="{expected_fragment}" (for path: {expected_path})')
```

Comment:
I don't understand why did you modify this function.
It should stay the same.

Reply:
Hello @jpp-odoo 
Changes have been made, please check it. 
Thanks!!